src/server.c

```diff
@@ -1837,10 +1837,9 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
 
     /* Run a fast expire cycle (the called function will return
      * ASAP if a fast cycle is not needed). */
+    long long expire_cycle_time = 0;
     if (server.active_expire_enabled && !server.import_mode && iAmPrimary()) {
```

Comment:
Don't we need to skip the expire work if ProcessingEventsWhileBlocked, to make sure the active time is summed correctly?

```suggestion
    if (server.active_expire_enabled &&
        !server.import_mode &&
        !ProcessingEventsWhileBlocked &&
        iAmPrimary()) {
```

Reply:
As I see it this shouldn't matter because of `!server.import_mode`, as we figured out - `ProcessingEventsWhileBlocked = true` for AOF/RDB loading, FULL sync, long scripts or long module cmd exec. In last two cases looks like we should expire any way and it would not be a good idea to skip expiration for them. 
If we add `!ProcessingEventsWhileBlocked` then long exec script/module cmd will get more lazy expirations.

To fix double accounting we can do:
```c
if (server.active_expire_enabled && !server.import_mode && iAmPrimary()) {
    long long expire_cycle_time = activeExpireCycle(ACTIVE_EXPIRE_CYCLE_FAST);
    if (ProcessingEventsWhileBlocked || !server.el_iteration_active) {
        server.stat_active_time += expire_cycle_time;
    }
}
```

Reply:
BTW there's also `activeExpireCycle()` call in `databasesCron()`, looks like it would be a good idea to count that time too, WDYT?

Reply:
Cron is a timer event in the event loop, so I would assume it's counted as active bacause of that.

I don't know why you think it's important to do incemental expiration work when a busy script is running. If it's slow, then interleaving more work would just make it even slower, right? Anyway, I don't think it matters much in practice, but in theory it's bad if we do double accounting in any case so let's fix that.

Reply:
> To fix double accounting we can do:
> 
> ```c
> if (server.active_expire_enabled && !server.import_mode && iAmPrimary()) {
>     long long expire_cycle_time = activeExpireCycle(ACTIVE_EXPIRE_CYCLE_FAST);
>     if (ProcessingEventsWhileBlocked || !server.el_iteration_active) {
>         server.stat_active_time += expire_cycle_time;
>     }
> }
> ```

Yes, this is a good idea!

Reply:
> I don't know why you think it's important to do incemental expiration work when a busy script is running.

TBH my first thought was just that I don’t want to break anything not related to CPU metric, but if you’re sure that nothing bad is going to happen if we skip incremental expiration on `ProcessingEventsWhileBlocked`, then we can add that too